feature-libs/subscription-billing/components/actions-modal/subscription-actions-modal.component.ts

```diff
@@ -80,14 +104,22 @@ export class SubscriptionActionsModalComponent {
   };
   constructor() {
     this.registerSubscriptionCancellationEffect();
+    const frequency = this.subscriptionContractFrequency;
+    const maxOptions: number =
+      (this.extendFrequencyMaxOptions[frequency] ?? 0) + 1; // +1 to include the unlimited duration option
+    this.extendDurationOptions = Array.from({ length: maxOptions }, (_, i) =>
+      i + 1 === maxOptions
+        ? this.UNLIMITED_DURATION
+        : (i + 1).toString() + ' ' + this.subscriptionContractFrequency
+    );
```

Comment:
`major:` let's extract this logic to a separate protected method so customers can customise its logic, if needed.